### PR TITLE
Muon Analysis 2: Output results in a table workspace that aren't numbers

### LINKED ISSUE #26180: Muon Analysis 2 results table type mismatch bug
### Expected behavior
Results table tab should be able to output run_start into a table workspace for all date-time formats
### Actual behavior
When an instroment outputs run_time in the format yyyy-mm-dd T hh:mm:ss it returns an error
### Steps to reproduce the behavior
1. Open workbench
1. Open Muon Analysis 2
1. Load MUSR 62260
1. Go to fitting tab 
1. Fit an Abragam function
1. Go to results tab
1. Check run_start
1. Output table
### Platforms affected
Windows

### PR DESCRIPTION
**Pull request by Pasarus**
**Description of work.**
Muon Analysis 2 had an issue with plotting a table workspace where a value was a string that was not convertible to a float. This now returns a string.

**To test:**

- Open workbench
- Open Muon Analysis 2
- Load MUSR 62260
- Go to fitting tab
- Fit an Abragam function
- Go to results tab
- Check run_start
- Output table

Fixes #26180. 

NEW FEATURE BUG FIX SO NO RELEASE NOTES

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
should check that the functional tests from this PR pass with the new GUI: 

https://github.com/mantidproject/mantid/pull/26200

File: scripts/Muon/GUI/Common/contexts/fitting_context.py [hunk 1]
```diff
@@ -268,13 +268,23 @@ def value_from_workspace(wksp_name):
...
         ]
-        return np.mean(values)
+        try:
+            return np.mean(values)
+        except TypeError:
+            # This will be a string
+            if len(values) == 1:
+                return values[0]
+            elif len(values) > 1:
+                return str(values[0]) + " to " + str(values[len(values) - 1])
```
Comment (Mantid-Matthew) [inline scripts/Muon/GUI/Common/contexts/fitting_context.py]: In python negative indices count from the end of an iterable so I think 
`return str(values[0]) + " to " + str(values[-1])` will work here.

Comment (Pasarus) [inline scripts/Muon/GUI/Common/contexts/fitting_context.py]:
```suggestion
                return str(values[0]) + " to " + str(values[-1])
```

**Review by Matthew-Andrew — APPROVED** [review]
Code changes look good and works for me :+1:

File: scripts/test/Muon/results_tab_widget/results_tab_model_test.py [hunk 2]
```diff
@@ -23,10 +24,9 @@
 
 
 def create_test_workspace(ws_name=None):
-    fake_ws = WorkspaceFactory.create('Workspace2D', 1, 1, 1)
     ws_name = ws_name if ws_name is not None else 'results_tab_model_test'
-    AnalysisDataService.Instance().addOrReplace(ws_name, fake_ws)
-    return fake_ws
+    Load("MUSR22725", OutputWorkspace=ws_name)
```
Comment (martyngigg) [inline scripts/test/Muon/results_tab_widget/results_tab_model_test.py]: I won't hold up merging this but does this really need to load data - it slows down the tests a lot?

**Pull request merged**